### PR TITLE
Add styling to user profile

### PR DESCRIPTION
**Pull request by TraceyOnim**

### CODE REVIEW

File: lib/hexpm_web/templates/layout/header.html.eex [hunk 1]
```diff
@@ -78,23 +78,42 @@
...
+                      <div class="-rotate-180">
+                        <%= icon(:remixicon, :"arrow-up-s-line", width: "16px", height: "16px", fill: "rgba(255,255,255,1)")%>
+                      </div>
+                    </div>
+                  </button>
+                  <!-- Dropdown menu -->
+                  <div class="hidden bg-white text-base z-50 rounded-lg h-36 w-40" id="dropdown">
+                    <ul class="py-6" aria-labelledby="dropdown">
+                      <li class="inline-flex px-6 font-medium text-sm text-gray-500">
+                       <%= icon(:remixicon, :"user-smile-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```
Comment (ericmj) [inline lib/hexpm_web/templates/layout/header.html.eex:98]:
```suggestion
                        <%= icon(:remixicon, :"user-smile-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```

File: lib/hexpm_web/templates/layout/header.html.eex [hunk 2]
```diff
@@ -78,23 +78,42 @@
...
+                        <a href="<%= Routes.user_path(Endpoint, :show, @current_user) %>" class="px-2">My Profile</a>
+                      </li>
+                      <li class="inline-flex px-6 font-medium text-sm text-gray-500">
+                          <%= icon(:remixicon, :"building-4-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```
Comment (ericmj) [inline lib/hexpm_web/templates/layout/header.html.eex:102]:
```suggestion
                        <%= icon(:remixicon, :"building-4-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```

File: lib/hexpm_web/templates/layout/header.html.eex [hunk 3]
```diff
@@ -78,23 +78,42 @@
...
+                        <a href="#" class="px-2">Organization</a>
+                      </li>
+                      <li class="inline-flex px-6 font-medium text-sm text-gray-500">
+                        <%= icon(:remixicon, :"settings-3-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
+                        <a href="<%= Routes.profile_path(Endpoint, :index) %>" class="px-2">Settings</a>
+                      </li>
+                      <li class="inline-flex px-6">
+                         <%= icon(:remixicon, :"logout-box-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```
Comment (ericmj) [inline lib/hexpm_web/templates/layout/header.html.eex:110]:
```suggestion
                        <%= icon(:remixicon, :"logout-box-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```

File: lib/hexpm_web/views/view_helpers.ex [hunk 4]
```diff
@@ -332,20 +340,48 @@ defmodule HexpmWeb.ViewHelpers do
...
+      month = round(year * 12)
+      if month == 1, do: "1 month ago", else: "#{month} months ago"
+    else
+      if round(year) == 1, do: "1 year ago", else: "#{round(year)} months ago"
+    end
+  end
+
+  defp no_of_days(datetime) do
+    date = DateTime.to_date(datetime)
+    Date.days_in_month(date)
```
Comment (ericmj) [inline lib/hexpm_web/views/view_helpers.ex:387]:
```suggestion
    datetime
    |> DateTime.to_date()
    |> Date.days_in_month()
```

File: lib/hexpm_web/templates/user/show.html.eex [hunk 5]
```diff
@@ -1,37 +1,141 @@
...
+          <%= for {icon, url} <- UserHandles.render(@user) do %>
+            <%= if icon == :elixirforum or icon == :freenode do%>
+              <a href="<%= url %>"><%= UserHandles.handle_svg_icon(icon) %></a>
+            <% else %>
+              <a href="<%= url %>"><%= icon(:remixicon, :"#{icon}", width: "16px", fill: "rgba(48, 66, 84, 1)")%></a>
+            <% end %>
+          <% end %>
+        </div>
+        <div class="border-t border-gray-200 pt-6">
+          <div class="grid grid-cols-2 flex flex-row gap-4">
```
Comment (ericmj) [inline lib/hexpm_web/templates/user/show.html.eex:26]: We can't have both grid and flex, they apply the same `display` CSS property.

**Review by ericmj — COMMENTED** [review]
I pushed a commit that fixes the number displays. It looks good overall but there's a few minor tweaks that needs to be done.

The icons are not aligned with the text:
Screenshot 2021-12-17 at 12 12 42
Screenshot 2021-12-17 at 12 12 47

The spacing/padding inside the box is larger at the top than at the bottom:
Screenshot 2021-12-17 at 12 13 07

There should be a shadow behind the dropdown so it stands out from the background. We also need to show the item that is being mouse overed:
Screenshot 2021-12-17 at 12 46 54

File: lib/hexpm_web/templates/layout/footer.html.eex [hunk 6]
```diff
@@ -67,6 +67,7 @@
 
 
   <script src="<%= Routes.static_path(Endpoint, "/assets/app.js") %>"></script>
+  <script src="https://unpkg.com/@themesberg/flowbite@latest/dist/flowbite.bundle.js"></script>
```
Comment (ericmj) [inline lib/hexpm_web/templates/layout/footer.html.eex:70]: What is this used for?

Comment (TraceyOnim) [reply lib/hexpm_web/templates/layout/footer.html.eex:70]: For the dropdown component, I opted to use https://flowbite.com/docs/getting-started/quickstart/ tailwindcss library. I used the CDN option to add it to the project

File: lib/hexpm_web/templates/user/show.html.eex (hunk 5 above)
Comment (TraceyOnim) [reply lib/hexpm_web/templates/user/show.html.eex:26]: Thank you for catching that , I should have removed the flex property

**Comment by TraceyOnim** [comment]
> I pushed a commit that fixes the number displays. It looks good overall but there's a few minor tweaks that needs to be done.
> 
> The icons are not aligned with the text: Screenshot 2021-12-17 at 12 12 42 Screenshot 2021-12-17 at 12 12 47
> 
> The spacing/padding inside the box is larger at the top than at the bottom: Screenshot 2021-12-17 at 12 13 07
> 
> There should be a shadow behind the dropdown so it stands out from the background. We also need to show the item that is being mouse overed: Screenshot 2021-12-17 at 12 46 54

Thank you ! Working on the corrections

File: lib/hexpm_web/templates/user/show.html.eex [hunk 7]
```diff
@@ -1,37 +1,137 @@
...
+            </div>
+          </div>
+        <% else %>
+          <div class="flex justify-between">
+            <div class="pt-2">
+              <h3 class="text-xl font-semibold text-gray-900">Packages</h3>
+            </div>
+            <div class="order-last pb-5">
+              <button class="border-box border border-solid border-gray-200 rounded-lg text-sm px-2 py-1 items-center justify-center" type="button" data-dropdown-toggle="package-dropdown">
+                <div class="flex inline-flex items-center space-x-2">
```
Comment (ericmj) [inline lib/hexpm_web/templates/user/show.html.eex:78]: `flex` and `inline-flex` set the same property.

File: lib/hexpm_web/templates/user/show.html.eex [hunk 8]
```diff
@@ -1,37 +1,137 @@
...
+                    <%= icon(:remixicon, :"arrow-down-s-line", width: "14px", height: "14px", fill: "rgba(97, 117, 138, 1)")%>
+                  </div>
+                </div>
+              </button>
+            </div>
+          </div>
+          <!-- Dropdown menu -->
+          <div class="hidden bg-white text-base z-50 rounded-lg h-28 w-40 shadow-2xl" id="package-dropdown">
+            <ul class="space-y-2 py-4" aria-labelledby="dropdown">
+              <li class="px-4 font-semibold text-sm text-gray-700">
```
Comment (ericmj) [inline lib/hexpm_web/templates/user/show.html.eex:90]: We shouldn't hardcode one as bold.

```suggestion
              <li class="px-4 font-medium text-sm text-gray-700">
```

File: lib/hexpm_web/templates/user/show.html.eex [hunk 9]
```diff
@@ -1,37 +1,137 @@
...
+                  <div class="font-medium text-sm text-gray-400">Most Popular</div>
+                  <div>
+                    <%= icon(:remixicon, :"arrow-down-s-line", width: "14px", height: "14px", fill: "rgba(97, 117, 138, 1)")%>
+                  </div>
+                </div>
+              </button>
+            </div>
+          </div>
+          <!-- Dropdown menu -->
+          <div class="hidden bg-white text-base z-50 rounded-lg h-28 w-40 shadow-2xl" id="package-dropdown">
```
Comment (ericmj) [inline lib/hexpm_web/templates/user/show.html.eex:88]: This dropdown should have the same design as the user dropdown on hover.

File: lib/hexpm_web/templates/layout/header.html.eex [hunk 10]
```diff
@@ -78,23 +78,42 @@
...
+                      <div class="-rotate-180">
+                        <%= icon(:remixicon, :"arrow-up-s-line", width: "16px", height: "16px", fill: "rgba(255,255,255,1)")%>
+                      </div>
+                    </div>
+                  </button>
+                  <!-- Dropdown menu -->
+                  <div class="hidden bg-white text-base z-50 rounded-lg h-36 w-40 shadow-2xl" id="dropdown">
+                    <ul class="py-6" aria-labelledby="dropdown">
+                      <li class="dropdown-menu">
+                        <%= icon(:remixicon, :"user-smile-line", width: "16px", fill: "rgba(68, 86, 104, 1)")%>
```
Comment (ericmj) [inline lib/hexpm_web/templates/layout/header.html.eex:98]: How about if the icons also change color on hover?

**Pull request merged**